src/cpp/include/openvino/genai/vlm_pipeline.hpp

```diff
@@ -0,0 +1,173 @@
+    // An encoder to infer embeddings of an image.
+    VisionEncoder m_vision_encoder;
+    // A resampler model to resample image embeddings.
+    // [N, H*W, old_hidden_size] is the input shape.
+    // [N, query_num, hidden_size] is the output shape.
+    ov::InferRequest m_resampler;
+    // A model to compute token embeddings.
+    // Input shape: [N, conversation length].
+    // Output shape: [1, conversation length, hidden_size].
+    ov::InferRequest m_embedding;
+    // A language model used to generate a response.
+    // Input shapes: inputs_embeds[N, conversation length, hidden_size],
+    // position_ids[N, conversation length], beam_idx[N].
+    // Output shape: logits[N, conversation length, vocab_size].
+    ov::InferRequest m_language;
+    // Precomputed positional embeddings for the resampler.
+    // [70, 70, hidden_size]. 70 is the initial guess of the image
+    // height and width after dividing by patch_size.
+    ov::Tensor m_pos_embed_cache;
+    // True if chat mode is activated to save conversation
+    // history between generate() calls.
+    bool m_is_chat_conversation;
+    ChatHistory m_history;
+    std::string m_templated_chat_history;
+    size_t image_id = 0;  // Used to insert <image_id>i</image_id> per image (not a slice).
+
+    /// @brief Construct a pipeline form a folder containing tokenizer
+    /// and model IRs.
+    /// @param model_dir A folder to read tokenizer and model IRs.
+    /// @param device Inference device. A tokenizer is always compiled
+    /// for CPU.
+    /// @param device_config A config to pass to ov::Core.set_property()
+    /// and ov::Core::compile_model().
+    /// @param core ov::Core instance to use.
+    explicit VLMPipeline(
+        const std::filesystem::path& model_dir,
+        const std::string& device="CPU",
+        const ov::AnyMap device_config={},
+        ov::Core core=ov::Core{}
+    ) : VLMPipeline{
+        model_dir,
+        Tokenizer(model_dir.string(), device_config),
+        device,
+        device_config,
+        core
+    } {}
+
+    /// @brief Construct a pipeline form a folder containing model IRs
+    /// and from a Tokenizer instance.
+    /// @param model_dir A folder to read model IRs.
+    /// @param tokenizer An instance of Tokenizer to use.
+    /// @param device Inference device.
+    /// @param device_config A config to pass to ov::Core.set_property()
+    /// and ov::Core::compile_model().
+    /// @param core ov::Core instance to use.
+    VLMPipeline(
+        const std::filesystem::path& model_dir,
+        const ov::genai::Tokenizer& tokenizer,
+        const std::string& device="CPU",
+        const ov::AnyMap device_config={},
+        ov::Core core=ov::Core{}
+    );
+
+    /// @brief Default destructor.
+    ~VLMPipeline();
+
+    /// @brief Generate a response given a prompt and any number of
+    /// uint8 RGB images.
+    /// @param prompt A prompt to respond to.
+    /// @param images Images to be prepended to a prompt.
```

Comment:
should we also mention that image is expected to be RGB NCHW ? (I suppose we don't perform auto-conversion from NHWC)

Reply:
resolved in https://github.com/Wovchena/openvino.genai-public/pull/68